Improve find command

lyueyang opened the pull request.

lyueyang committed:
Improve code quality

- Remove magic strings
- Move parsing to ParserUtil

lyueyang committed:
Change arguments to lowercase

Done to accommodate multi-word search using contains

lyueyang committed:
Add multi-word search for attributes with spaces

Attributes such as address and names which can have multiple words due to spaces can be searched in whole

lyueyang committed:
Add description for FindCommand behaviour

lyueyang committed:
Address incorrect info

lyueyang committed:
Update UG for clarity

Unable to show multiple spaces with inline code block with github markdown

lyueyang committed:
Merge remote-tracking branch 'AY2021S2-CS2103T-W10-1/master' into improve-find-command

lyueyang committed:
Remove more magic literals

ZechariahTan commented on src/main/java/seedu/address/logic/parser/ParserUtil.java:
```diff
@@ -209,4 +213,41 @@ public static Price parsePrice(String price) throws ParseException {
+                    break;
+                }
+
+            } catch (ParseException ignored) {
```
I don't know if you can do it like that? I might be wrong but isn't leaving an empty catch block not good coding practice?

ZechariahTan replied:
Ok my mistake looks like parseKeyword from the original AB3 does something similar.

ZechariahTan replied:
maybe just insert a continue?

ZechariahTan commented on src/test/java/seedu/address/model/person/passenger/NameContainsKeywordsPredicateTest.java:
```diff
@@ -47,20 +50,23 @@ public void equals() {
 
         // Mixed-case keywords
-        predicate = new NameContainsKeywordsPredicate(Arrays.asList("aMy", "beE"));
+        // Keywords changed to lowercase as parser will change all keywords to lowercase
```
Maybe add this comment to the earlier testcases also? might make it more clear why everything has been made to lowercase.

lyueyang committed:
Add comment explaining rationale for lowercase

- Keywords are converted to lowercase to as Parser does the conversion for predicates that expect strings

lyueyang committed:
Refine comment for empty catch block

lyueyang committed:
Fix merge conflict

ZechariahTan reviewed (approved):
LGTM
